src/pokeradar.c

```diff
@@ -0,0 +1,590 @@
+            chain->patch[patchRing].unk_18 = NULL;
+        }
+    }
+}
+
+BOOL sub_02069690 (RadarChain * chain)
+{
+    u8 v0;
+    u8 patchRing;
+
+    v0 = 0;
+
+    for (patchRing = 0; patchRing < NUM_GRASS_PATCHES; patchRing++) {
+        if (chain->patch[patchRing].unk_18 != NULL) {
+            if (ov5_021F31A8(chain->patch[patchRing].unk_18)) {
+                sub_0207136C(chain->patch[patchRing].unk_18);
+                chain->patch[patchRing].unk_18 = NULL;
+                v0++;
+            }
+        } else {
+            v0++;
+        }
+    }
+
+    if (v0 >= 4) {
+        return 1;
+    }
+
+    return 0;
+}
+
+BOOL sub_020696DC (const int param0, const int param1, UnkStruct_0203CDB0 * param2, RadarChain * chain, int * param4, BOOL * param5, BOOL * param6)
+{
+    u8 patchRing;
+    BOOL continueChain;
+    int shakeType;
+
+    *param5 = 0;
+    *param6 = 0;
+
+    if (!sub_020698AC(chain, param0, param1, &patchRing)) {
+        return 0;
+    }
+
+    chain->unk_18 = 1;
+
+    continueChain = chain->patch[patchRing].continueChain;
+    shakeType = chain->patch[patchRing].shakeType;
+
+    if (chain->unk_14 == 0) {
+        if (continueChain) {
+            IncWithCap(&(chain->count));
+
+            *param4 = shakeType;
+            *param5 = 1;
+
+            sub_020698E4(param2, chain);
+
+            (*param6) = chain->patch[patchRing].shiny;
+
+            return 1;
+        } else {
+            *param4 = shakeType;
+        }
+    } else {
+        *param4 = shakeType;
+        chain->unk_14 = 0;
+
+        chain->unk_D0 = sub_0206994C(param2);
+    }
+
+    chain->shakeType = *param4;
+
+    return 1;
+}
+
+void SetRadarMon (RadarChain * chain, const int species, const int level) {
+    GF_ASSERT(species != 0);
+    chain->species = species;
+    chain->level = level;
+}
+
+void GetRadarMon (RadarChain * chain, int * species, int * level) {
+    (*species) = chain->species;
+    (*level) = chain->level;
+}
+
+const BOOL sub_02069798 (const RadarChain * chain)
+{
+    return chain->unk_18;
+}
+
+void sub_0206979C (UnkStruct_0203CDB0 * param0)
+{
+    BOOL v0;
+    GrassPatch * patch;
+    int patchRing;
+
+    if ((!param0->chain->active) || (param0->unk_10 != NULL)) {
+        return;
+    }
+
+    for (patchRing = 0; patchRing < NUM_GRASS_PATCHES; patchRing++) {
+        patch = &(param0->chain->patch[patchRing]);
+        v0 = sub_0201CF7C(&(patch->position), &(param0->chain->unk_BC));
+
+        if ((patch->active) && (!v0)) {
+            patch->active = FALSE;
+        }
+    }
+
+    {
+        int v3;
+
+        v3 = 0;
+
+        for (patchRing = 0; patchRing < NUM_GRASS_PATCHES; patchRing++) {
+            patch = &(param0->chain->patch[patchRing]);
+
+            if (patch->active == 0) {
+                v3++;
+            }
+        }
+
+        if (v3 == 4) {
+            ClearRadarChain(param0->chain);
+
+            sub_02055554(param0, sub_02055428(param0, param0->unk_1C->unk_00), 1);
```

Comment:
Does this need to be in a separate block to match? Try to reduce overall indentation as much as possible.